src/main/java/com/comphenix/protocol/injector/EntityUtilities.java

```diff
@@ -203,40 +204,59 @@ private Collection<?> getTrackedPlayers(Entity entity) {
 
     @SuppressWarnings("unchecked")
     private Object getNewEntityTracker(Object worldServer, int entityId) {
-        if (this.getChunkProvider == null) {
-            Class<?> chunkProviderClass = MinecraftReflection.getChunkProviderServer();
-            this.getChunkProvider = Accessors.getMethodAccessor(FuzzyReflection.fromClass(worldServer.getClass(), false)
-                    .getMethod(FuzzyMethodContract.newBuilder().parameterCount(0).returnTypeExact(chunkProviderClass).build()));
-        }
+        if (Util.isUsingFolia()) {
+            if (this.getEntity == null) {
+                Method entityGetter = FuzzyReflection.fromObject(worldServer).getMethodByReturnTypeAndParameters(
+                        "getEntity",
+                        MinecraftReflection.getEntityClass(),
+                        int.class);
+                this.getEntity = Accessors.getMethodAccessor(entityGetter);
+            }
 
-        Object chunkProvider = this.getChunkProvider.invoke(worldServer);
+            Object entity = this.getEntity.invoke(worldServer, entityId);
 
-        if (this.chunkMapField == null) {
-            Class<?> chunkMapClass = MinecraftReflection.getPlayerChunkMap();
-            this.chunkMapField = Accessors.getFieldAccessor(FuzzyReflection.fromClass(chunkProvider.getClass(), false)
-                    .getField(FuzzyFieldContract.newBuilder().typeExact(chunkMapClass).build()));
-        }
+            if (this.foliaTrackerField == null) {
+                this.foliaTrackerField = Accessors.getFieldAccessor(FuzzyReflection.fromClass(entity.getClass(), false)
+                        .getField(FuzzyFieldContract.newBuilder().typeExact(MinecraftReflection.getEntityTrackerClass()).build()));
+            }
 
-        Object playerChunkMap = this.chunkMapField.get(chunkProvider);
+            return this.foliaTrackerField.get(entity);
+        } else {
```

Comment:
I don't *think* ProtocolLib follows any min diff rules, but if you just did the
```java
if (isFolia) {
    // logic here
    return;
}

// the else logic goes here
```
You can avoid changing the indentation of the original code. Usually a good rule of thumb when modifying other people's code.